From 5e566d1877dfeb578bc4a410d19fb7a0a1abe3f5 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:13 -0400
Subject: [block] blk-mq: simplify blk_mq_hw_sysfs_cpus_show()

Message-id: <1402670238-13916-26-git-send-email-snitzer@redhat.com>
Patchwork-id: 83871
O-Subject: [RHEL7.1 PATCH 025/150] blk-mq: simplify blk_mq_hw_sysfs_cpus_show()
Bugzilla: 1105204
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105204

Upstream commit cb2da43e3d02b745408252fe4c447b23b1f4fc7b
Author: Jens Axboe <axboe@fb.com>
Date:   Wed Apr 9 10:53:21 2014 -0600

    blk-mq: simplify blk_mq_hw_sysfs_cpus_show()

    Now that we have a cpu mask of CPUs that are mapped to
    a specific hardware queue, we can just iterate that to
    display the sysfs num-hw-queue/cpu_list file.

    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq-sysfs.c b/block/blk-mq-sysfs.c
index b0ba264..9176a69 100644
--- a/block/blk-mq-sysfs.c
+++ b/block/blk-mq-sysfs.c
@@ -246,16 +246,12 @@ static ssize_t blk_mq_hw_sysfs_tags_show(struct blk_mq_hw_ctx *hctx, char *page)
 
 static ssize_t blk_mq_hw_sysfs_cpus_show(struct blk_mq_hw_ctx *hctx, char *page)
 {
- unsigned int i, queue_num, first = 1;
+ unsigned int i, first = 1;
  ssize_t ret = 0;
 
  blk_mq_disable_hotplug();
 
- for_each_online_cpu(i) {
-  queue_num = hctx->queue->mq_map[i];
-  if (queue_num != hctx->queue_num)
-   continue;
-
+ for_each_cpu(i, hctx->cpumask) {
   if (first)
    ret += sprintf(ret + page, "%u", i);
   else
-- 
1.7.1